beacon/beacon.go

```diff
@@ -358,7 +358,7 @@ func (pd *ProtocolDriver) MinerAllowance(epoch types.EpochID, nodeID types.NodeI
 	if err != nil || malicious {
 		return 0
 	}
-	return atx.NumUnits
+	return atx.EffectiveNumUnits
```

Comment:
`ActivationTxHeader` is just like `atxdata.ATX` basically the same as the domain type now.

One option would be to now get rid of them and replace them with the now smaller `types.ActivationTx`.

The second option would be to have a "`types.BeaconATX`" that only contains the fields that the beacon is actually interested in and a dedicated function in the `sql/atxs` package like `atxs.BeaconATX(id types.ATXID)` (or similar) to query the relevant data from the DB and avoiding to first have to query a `types.ActivationTx` and then copy over only the relevant data for beacon into a new (smaller) object.

Wdyt?

Reply:
I eventually want to get rid of the `ActivationTxHeader`, just after merging `VerifiedActivationTx` with `ActivationTx`. 

I'm not yet sure about `atxdata.ATX` type. It's considerably smaller because it doesn't contain many big fields (i.e. PrevATXID and id). Given we store **in memory** about 2 full epochs of `atxdata.ATX`, it accumulates into a few hundred MiB saved.

Reply:
I think after cleaning up everything `types.ActivationTx` won't be much bigger than `atxdata.ATX` and we could even further reduce the memory footprint by just having more specific types in the services that only need some of the fields that `types.ActivationTx` holds.